mms/tests/unit_tests/test_utils/test_codec_helpers.py

```diff
@@ -16,3 +18,36 @@ def test_invalid_encode(self):
         msg = b"Hello World!!"
         with pytest.raises(MMSError):
             ModelWorkerCodecHelper.encode_msg('dummy', msg)
+
+    class TestDecodeMsg():
+        encoded_msg = "SGVsbG8gV29ybGQh"
+
+        def test_decode_msg(self):
+            assert b"Hello World!" == ModelWorkerCodecHelper.decode_msg('base64', self.encoded_msg)
+
+        def test_with_no_encoding(self):
+            assert self.encoded_msg == ModelWorkerCodecHelper.decode_msg('garbage', self.encoded_msg)
+
+        def test_with_error(self, mocker):
+            b64_mock = mocker.patch('base64.b64decode')
+            exception = binascii.Error('error')
+            b64_mock.side_effect = exception
+
+            with pytest.raises(Exception) as ex:
+                ModelWorkerCodecHelper.decode_msg('base64', self.encoded_msg)
+
+            assert isinstance(ex.value, MMSError)
+            assert ex.value.get_code() == err.DECODE_FAILED
+            assert ex.value.get_message() == "base64 decode error {}".format(repr(exception))
```

Comment:
can't str() do this?

Reply:
They are slightly different.  For exceptions, str(Exception('msg')) == 'msg' and repr(Exception('msg')) == "Exception('msg')"

Reply:
Here's the difference : 
str(exception) prints just the message part of the exception.

repr(exception) prints the exception message as well as the exception class.